### PR TITLE
BaSP-TG-114: Employee form update

### PR DESCRIPTION
**Pull request by ProDIGGY-vinilos**
Update the logic in the employee entity to use rect hook form and have joi validations

-  How to test:
- [ ] Check if the employee entity works properly, all inputs have the corrects validations and the errors are displayed correctly

### CODE REVIEW

File: src/Components/Employees/EmployeeForm.js [hunk 1]
```diff
@@ -58,49 +61,45 @@ const EmployeeForm = () => {
...
   return (
     <>
-      <form className={styles.container} onSubmit={onSubmit}>
+      <form className={styles.container} onSubmit={handleSubmit(onSubmit)}>
         <h3 className={styles.titleForm}>Create Employee</h3>
         <Button href="/employees" style="roundedSecondary" disabled={false} text="X" />
         <div className={styles.formControl}>
           <InputField
-            label="Name"
+            label="Namea"
```
Comment (MaicolFreire001) [inline src/Components/Employees/EmployeeForm.js:97]: You misspelled here

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/Components/Employees/EmployeeForm.js (hunk 1 above)
Comment (ProDIGGY-vinilos) [reply src/Components/Employees/EmployeeForm.js:97]: Uh, sorry i added that to see the input with that identation, done

**Review by MaicolFreire001 — APPROVED** [review]
SGTM🚀

File: src/Components/Employees/validation.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Joi from 'joi';
+
+export const schema = Joi.object({
+  name: Joi.string()
+    .alphanum()
+    .pattern(/^([^0-9]*)$/i, 'only letters')
+    .required(),
```
Comment (ignacioaristo) [inline src/Components/Employees/validation.js:7]: custom the message of this validation

File: src/Components/Employees/EmployeeForm.js [hunk 3]
```diff
@@ -58,38 +61,35 @@ const EmployeeForm = () => {
     }
   }, [error]);
 
+  const MOCK_DATA = {
```
Comment (ignacioaristo) [inline src/Components/Employees/EmployeeForm.js:64]: Change this name, this is not a data mocked. Is real data

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

**Review by JuliMonte5 — APPROVED** [review]

File: src/Components/Employees/EmployeeForm.js (hunk 3 above)
Comment (Ignacio-Sormani) [reply src/Components/Employees/EmployeeForm.js:64]: Also it should be in camelCase

Comment (ProDIGGY-vinilos) [reply src/Components/Employees/EmployeeForm.js:64]: Sorry, i followed the class too literally

File: src/Components/Employees/validation.js (hunk 2 above)
Comment (ProDIGGY-vinilos) [reply src/Components/Employees/validation.js]: Done

File: src/Components/Employees/EmployeeForm.js (hunk 3 above)
Comment (ProDIGGY-vinilos) [reply src/Components/Employees/EmployeeForm.js]: done

**Review by sofihoj — APPROVED** [review]

File: src/Components/Employees/validation.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import Joi from 'joi';
+
+export const schema = Joi.object({
```
Comment (Ignacio-Sormani) [inline src/Components/Employees/validation.js:3]: We should have messages for every input

Comment (ProDIGGY-vinilos) [reply src/Components/Employees/validation.js:3]: Done

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**